Cabal/Distribution/Simple/Command/Test.hs

```diff
@@ -0,0 +1,188 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Distribution.Simple.Command.Test
+-- Copyright   :  Thomas Tuegel 2016
+-- License     :  BSD3
+--
+-- Maintainer  :  cabal-devel@haskell.org
+-- Portability :  portable
+--
+-- This module defines command-line interface to the @test@ action.
+
+module Distribution.Simple.Command.Test where
+
+import GHC.Generics ( Generic )
+import qualified Text.PrettyPrint as Disp
+
+import qualified Distribution.Compat.ReadP as Parse
+import Distribution.Compat.Semigroup
+import Distribution.Flag
+import Distribution.ReadE
+import Distribution.Simple.Command
+import Distribution.Simple.InstallDirs
+import Distribution.Simple.Utils
+import Distribution.Text
+import Distribution.Verbosity
+
+import Data.Char ( isAlpha )
+
+
+data TestConfig a = TestFlags { testDistPref    :: a FilePath
+                              , testVerbosity   :: a Verbosity
+                              , testHumanLog    :: a PathTemplate
+                              , testMachineLog  :: a PathTemplate
+                              , testShowDetails :: a TestShowDetails
+                              , testKeepTix     :: a Bool
+                              , testOptions     :: [PathTemplate]
```

Comment:
In the putative docs, there needs to be an explanation when you say `f MyType` and when you say `[MyType]`. I think the reason is that the default for a list argument is always `[]`, and then if you specify multiple times you want it to cons on, e.g. `--dependency foo --dependency bar`?